usr/lib/cca_stdll/cca_specific.c

```diff
@@ -6129,21 +6131,51 @@ CK_RV token_specific_aes_cbc(STDLL_TokData_t * tokdata,
     return rc;
 }
 
+static CK_BBOOL token_specific_filter_mechanism(STDLL_TokData_t *tokdata,
+                                                CK_MECHANISM_TYPE mechanism)
+{
+    CK_BBOOL rc = CK_FALSE;
+
+    switch(mechanism) {
+    case CKM_AES_XTS:
+    case CKM_AES_XTS_KEY_GEN:
+#ifndef NO_PKEY
+         if (ccatok_pkey_option_disabled(tokdata) &&
```

Comment:
This should be OR not AND:

```
if (ccatok_pkey_option_disabled(tokdata) ||
    !((struct cca_private_data *)tokdata->private_data)->pkey_wrap_supported) {
```
If either the option is disabled, or the cards does not support it, then we want to filter out the XTS mechs.